org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/synapse/connector/ConnectionHandlerTest.java

```
@@ -0,0 +1,114 @@
+import org.eclipse.lemminx.customservice.synapse.connectors.entity.Connector;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import static org.eclipse.lemminx.synapse.TestUtils.getResourceFilePath;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ConnectionHandlerTest {
+
+    ConnectionHandler connectionHandler;
+
+    @BeforeAll
+    public void setUp() throws URISyntaxException {
+
+        connectionHandler = new ConnectionHandler();
+        String connectorPath = getResourceFilePath("/synapse/connectors/extracted/mi-connector-http-0.1.8");
+        ConnectorReader connectorReader = new ConnectorReader();
+        Connector connection =
+                connectorReader.readConnector(connectorPath, null);
+        ConnectorHolder.getInstance().addConnector(connection);
+        connectionHandler.init(ConnectorHolder.getInstance());
+    }
+
+    @Test
+    public void testGetConnectionSchemaFromValidName() throws IOException {
+
+        JsonObject connection1Schema = connectionHandler.getConnectionUiSchema("http", "http");
+        JsonObject connection2Schema = connectionHandler.getConnectionUiSchema("http", "https");
+
+        assertValidConnectionSchema(connection1Schema, "HTTP");
+        assertValidConnectionSchema(connection2Schema, "HTTPS");
+    }
+
+    @Test
+    public void testGetConnectionSchemaFromInvalidName() throws IOException {
+
+        JsonObject connectionSchema = connectionHandler.getConnectionUiSchema("invalid", "invalid");
+        assertNull(connectionSchema);
+    }
+
+    @Test
+    public void testGetConnectionSchemaForValidConnectionFile() throws IOException, URISyntaxException {
```

We can simply throw an Exception right as we are not doing any action based on the exception thrown?